### DISCUSSION TITLE
25.01.rc3

### DISCUSSION THREAD

**Discussion by SPMcentral** [spm/spm#44] [General]
We are delighted to be sharing this release candidate for SPM 25.01. Please test thoroughly and get in touch via the Discussion forum if you find any issues.

**Updates since previous release candidate (25.01.rc2)**

- Updated author list

<hr /><em>This discussion was created from the release <a href='https://github.com/spm/spm/releases/tag/25.01.rc3'>25.01.rc3</a>.</em>

**Comment by CccAoo** [comment]
hi, everyone.
I have just found a question about SPM 25.01
when I creat a new batch include spm_jobman to  preprocessing script, I get the following error message: 
Error using spm_jobman (line 157)
Don't know how to handle this 'run' call.
who knows what' s wrong with this?
thanks.
Cc

**Reply by pzeidman** [reply]
Apologies we missed this question. I think you simply need to initialize SPM before calling spm_jobman . Try putting this at the top of your script:

```
spm('defaults','fmri');
spm_jobman('initcfg');
```

Change fmri to eeg if required.

Let us know if that resolves it for you,
Peter

**Comment by chiuhoward** [comment]
Hello, I was wondering if the C function signatures in this repository would need to be updated at some point in the future? 

For example,

https://github.com/spm/spm/blob/0ccbd8113ae1b69c7f710351130332b57a14f9fb/src/spm_vol_utils.c#L315-L318

to something like

```
void RESAMPLE_0(int m, IMAGE_DTYPE *vol[], double out[], double x[], double y[], double z[], 
                int xdim, int ydim, int zdim, double background, double scale[], double offset[]);
```

Thanks!

**Reply by balbasty** [reply]
Congratulations, you have found a fossilised piece of pre-ANSI C!

Have you had issues compiling that code? If no I would say "let's not fix what's not broken" :)

**Comment by zhanminye** [comment]
Hello, I'm trying out the scope top-up EPI distortion correction toolbox (spm_scope.m). 
The code produced the vdm and the unwarped images for single volumes, but not for a time series (multiple volumes). 
What is the proper way to apply the vdm to an EPI time series? 
Thanks!

**Reply by balbasty** [reply]
Hello,

You should use the "realign & unwarp" tool, which takes the SCOPE voxel displacement map as input.
Note that realign & unwarp does a bit more than simply unwarping using the field map, as it also estimates dynamic changes to the field caused by bulk motion.

There is currently no way of only applying the static field map, as far as I known, but the dynamic refinement should hopefully have very little effect in cases of little head motion.

You may want to compare the results with and witout jacobian modulation ("on" by default), as we've seen cases in which the jacobian modulation induces unwanted variance.

@JohnAshburner may have more to say.

**Comment by escholey** [comment]
Hi, I had a query about the Unwarp defaults in SPM 25. Jacobian modulation has now changed to 'on' by default. However, I've been getting strange intensity distortions when unwarping, which disappear when I turn Jacobian modulations off, see example below: 
<img width="257" alt="Screenshot 2025-03-09 at 11 53 19" src="https://github.com/user-attachments/assets/7917a29b-7fad-4d6e-bb60-daa3b110af0a" />

Reading the response to the previous question, it looks like jacobian modulation might be causing problems in some places - please could you advise what might be going on? 

Thanks!

**Comment by tierneytim** [comment]
Hi @escholey, 

Just transferring this discussion to an issue, It's better to raise problems there than here
